### PR TITLE
fuul sdk set up

### PR DESCRIPTION
**Pull request by Seroxdesign**
Closes: #456 

## Changes made:

1. Integrated the FUUL SDK
2. Created a hook which initializes the SDK, and exports functions to send events
3. Created sign event, which asks user to sign a message

### CODE REVIEW

File: src/components/Steps.tsx [hunk 1]
```diff
@@ -24,11 +25,24 @@ const Steps = () => {
...
+
     const handleCreateAccount = async () => {
         if (!account || !provider || !chainId) return false
-        const txData = await geb.contracts.proxyRegistry.populateTransaction['build()']()
-        const signer = provider.getSigner(account)
+
         try {
+            await sendConnectWalletEvent(account)
+            const txData = await geb.contracts.proxyRegistry.populateTransaction['build()']()
+            const signer = provider.getSigner(account)
```
Comment (pi0neerpat) [inline src/components/Steps.tsx:45]: Why is this moved inside the try-catch?

File: src/components/Steps.tsx [hunk 2]
```diff
@@ -24,11 +25,24 @@ const Steps = () => {
 
     const handleConnectWallet = () => popupsActions.setIsConnectorsWalletOpen(true)
 
+    const sendConnectWalletEvent = async (walletAddress: string): Promise<void> => {
```
Comment (pi0neerpat) [inline src/components/Steps.tsx:28]: Is this meant to be duplicated with the useFuulSdk hook?

**Review by pi0neerpat — CHANGES_REQUESTED** [review]

File: src/components/Steps.tsx (hunk 2 above)
Comment (Seroxdesign) [reply src/components/Steps.tsx:28]: It shouldn't be duplicated we should use the method in the hook @jahabeebs are you handling it or should I?

File: src/components/Steps.tsx (hunk 1 above)
Comment (jahabeebs) [reply src/components/Steps.tsx:45]: because sendConnectWalletEvent is a network call and signature request and both can fail

File: src/components/Steps.tsx (hunk 2 above)
Comment (jahabeebs) [reply src/components/Steps.tsx:28]: > It shouldn't be duplicated we should use the method in the hook @jahabeebs are you handling it or should I?

I fixed it

**Comment by jahabeebs** [comment]
@Seroxdesign @pi0neerpat I've added the following changes to this branch:
1) in App.tsx we now check for a referrer or af variable and save referralProgram === 'true' to localStorage if one of those are there
2) Signature requests now use sign in with ethereum and will depend on whether referralProgram === 'true'. I have it set up now that even if a user denies being in the points program it will still ask them to create a vault facilitator afterward (it won't block the vault facilitator process)
3) Added new methods to useFuulSDK.ts and added an affiliate page at the route /affiliate where a user can create an affiliate code, it then registers that code with the Fuul API, and the user's wallet address is now connected to that affiliate code (so it will be persisted thanks to the Fuul API) and the user will be able to copy and paste a referral link

Here are some screenshots:

Signature request when generating an affiliate code (note that it has to look like this, Fuul's SDK will reject the signature if we try to use sign in with ethereum or use different signature text)

https://github.com/open-dollar/od-app/assets/47253537/30e09aa3-4de7-4e19-8995-94e3c1abcd9c

Error when affiliate code already taken

https://github.com/open-dollar/od-app/assets/47253537/8840a768-ff95-4b28-b502-9433e3cc0d2d

Disabled when wallet not connected and when no text in input

https://github.com/open-dollar/od-app/assets/47253537/891cd97b-bc67-4827-9d9d-015565185448

Successful generation

https://github.com/open-dollar/od-app/assets/47253537/62316ecc-efbd-4390-8cb9-a8ceb0e1b007

Code remembered on reconnection

https://github.com/open-dollar/od-app/assets/47253537/2837bf3c-663d-4abe-b83c-26922fa1db09

File: src/containers/Affiliate/index.tsx [hunk 3]
```diff
@@ -0,0 +1,141 @@
...
+                                {copied ? 'Copied!' : 'Copy Affiliate Link'}
+                                <CopyIconContainer>
+                                    <CopyIcon />
+                                </CopyIconContainer>
+                            </Button>
+                        </CopyToClipboard>
+                    </>
+                ) : (
+                    <>
+                        <input
```
Comment (pi0neerpat) [inline src/containers/Affiliate/index.tsx:77]: Can you show the url on the left, and the input box on the right? This way its clear how the final result will look.

Here's how I imagine it

image

**Review by pi0neerpat — APPROVED** [review]
Incredible! Just one minor cosmetic change and its good to go!
